### ISSUE TITLE
False positive with redundant ignore and :clojure-lsp/unused-public-var

### ISSUE THREAD

**Issue by vincentjames501** [clj-kondo/clj-kondo#2467]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

```bash
$ clj-kondo --version
clj-kondo v2025.01.16
```

**platform**

native

**problem**

Clojure-lsp detects a few things that clj-kondo doesn't natively such as unused-public-var. With the latest release of clj-kondo, when we have something like this:

```clojure
#_{:clj-kondo/ignore [:clojure-lsp/unused-public-var]}
(defn foo
  [x]
  (* x 42))
```

It will produce something like:

```bash
$ clj-kondo --lint src
src/foo.clj:12:3: info: Redundant ignore
```

**expected behavior**

Ideally if clj-kondo recognize the linter being ignored it would skip the redundant ignore message. I do realize it's only an info level log though.

**Comment by vincentjames501** [comment]
I see I can tune this a bit with `--report-level`, just not w/in config.edn

**Comment by borkdude** [comment]
Maybe it's a good rule of thumb to disable redundant-ignore when the ignore hint contains namespaced keywords (which is a sign of a custom linter).

**Comment by vincentjames501** [comment]
> Maybe it's a good rule of thumb to disable redundant-ignore when the ignore hint contains namespaced keywords (which is a sign of a custom linter).

I think this makes sense. Would also be nice if we could set `:report-level` in `config.edn` since generally info level messages aren't important to our git pre-commit workflows or ci linting stages

**Comment by borkdude** [comment]
Another thought, perhaps it's better to add an exclude option to this linter, such that you can disable it yourself, since not all custom linters have to be disabled to prevent false positives.


``` clojure
:linters {:redundant-ignore {:exclude [:clojure-lsp/unused-public-var]}}
```

**Comment by vincentjames501** [comment]
@borkdude That would work for us!

**Comment by dgreensp** [comment]
I got here via web search just trying to figure out how to ignore `unused-public-var` in the first place...

If Kondo can ignore `unused-public-var` when asked to, why can't it tell if the ignore is redundant or not?  It seems the resolution here solves the immediate problem, but things are still not ideal.

I don't know the relationship between the Clojure LSP and Kondo.  I'm using VS Code/Calva, which invokes one which invokes the other, I assume.

**Comment by borkdude** [comment]
It uses to be possible but clojure-lsp no longer runs this liner as part of clj-kondo, but outside of it.

cc @ericdallo
